### PR TITLE
Use unique tags as key in the settings dictionary

### PR DESCRIPTION
**Pull request by AntonioLangiu**
See issue #522. 

To finish this PR I need to change the tag in the documentation.

### CODE REVIEW

File: src/net/connect.cpp [hunk 1]
```diff
@@ -110,11 +111,11 @@ void resolve_hostname(
             }
             cb(*result);
         }, {}, poller);
```
Comment (bassosimone) [inline src/net/connect.cpp]: Shouldn't this `{}` become `settings`? Or is this something to be done in another pull request?

Comment (AntonioLangiu) [inline src/net/connect.cpp]: yes, It should be done in another PR. 

In this PR I just changed the tags.

Comment (bassosimone) [inline src/net/connect.cpp]: ack
